cdap-app-fabric/src/main/java/co/cask/cdap/reporting/ProgramHeartbeatDataset.java

```diff
@@ -174,15 +168,14 @@ public void writeRunRecordMeta(RunRecordMeta runRecordMeta, long timestampInSeco
 
       // since the serialized runRecordMeta doesn't have programRunId (transient), we will create and
       // add the programRunId to RunRecordMeta and add to result list
-      List<RunRecordMeta> runRecordMetaList =
+      runRecordMetas.addAll(
```

Comment:
this still loops twice, once to build the list, then another time to add it. I think you can just do it with a normal for loop

Reply:
looks like i can use `forEachOrdered` and use the list add method to add to the result list. open to other suggestions.

Reply:
using `forEach` pattern instead of `stream` simplifies adding to result list. using that.